### DISCUSSION TITLE
Ui.button in loop does not accept arguments

### DISCUSSION THREAD

**Discussion by phneoix** [zauberzeug/nicegui#3332] [Q&A]
### Question

Hi,
   i am trying to create buttons in loop and facing a problem as below.
```py
xls = ['a','b','c','d']
        
for x in xls:
      ui.button(x,on_click=lambda:ui.notify(x))
```

the notification always shows 'd'.

**Comment by phneoix** [comment]
closing the thread. Found answer in #471.
```py
xls = ['a','b','c','d']

for x in xls:
      ui.button(x,on_click=lambda x=x:ui.notify(x))
```

**Reply by falkoschindler** [reply]
Exactly. It's a pretty common question: https://github.com/zauberzeug/nicegui/wiki/FAQs#why-do-all-my-elements-have-the-same-value.
